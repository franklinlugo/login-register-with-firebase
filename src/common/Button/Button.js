import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
  display: grid;
  place-items: center;
  width: 100%;
  height: 60px;
  background-color: ${(props) => props.theme.primary};
  color: ${(props) => props.theme.white};
  font-weight: 600;
  cursor: pointer;
  transition: 0.3s;
  transform-style: preserve-3d;
  &:active {
    transform: scale(0.97);
  }
`;

function Button({ children, type = 'button' }) {
  return <StyledButton type={type}>{children}</StyledButton>;
}

Button.propTypes = {
  children: string,
  type: string,
};

export default Button;
