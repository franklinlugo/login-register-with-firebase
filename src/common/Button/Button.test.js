import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

test('Button render correctly', () => {
  const { asFragment } = render(<Button>button</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test('Button handle type', () => {
  const { asFragment } = render(<Button type="submit">button</Button>);
  expect(asFragment()).toMatchSnapshot();
});
