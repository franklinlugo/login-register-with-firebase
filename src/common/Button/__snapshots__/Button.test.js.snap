// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button handle type 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm HosKC"
    type="submit"
  >
    button
  </button>
</DocumentFragment>
`;

exports[`Button render correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm HosKC"
    type="button"
  >
    button
  </button>
</DocumentFragment>
`;
